File: 백준/Bronze/3009. 네 번째 점/네 번째 점.js
const fs = require('fs');
const dots = fs.readFileSync(0, 'utf-8').toString().trim().split('\n').map(x => x.split(' ').map(Number));
const setX = new Set();
const setY = new Set();

for (let [x, y] of dots) {
    if (setX.has(x)) {
        setX.delete(x);
    } else {
        setX.add(x);
    }
    
    if (setY.has(y)) {
        setY.delete(y);
    } else {
        setY.add(y);
    }
}

console.log(Array.from(setX)[0], Array.from(setY)[0]);

File: 백준/Bronze/3009. 네 번째 점/네 번째 점.py
import sys

set_x = set()
set_y = set()

for row in sys.stdin.readlines():
    x, y = map(int, row.split())
    
    if x in set_x:
        set_x.remove(x)
    else:
        set_x.add(x)
        
    if y in set_y:
        set_y.remove(y)
    else:
        set_y.add(y)

print(f'{next(iter(set_x))} {next(iter(set_y))}')